### PR TITLE
[TEVA-2078] - use github container registery

### PR DESCRIPTION
**Pull request by felixtheflex**
## Jira ticket URL

- Just add the ticket number to the end:

https://dfedigital.atlassian.net/browse/TEVA-2078

## Changes in this PR:

Change from using Docker Registry to using Github packages for Docker and changing authentication to `ghcr.io` via `docker/login-action@v1` by using `${{ github.repository_owner }}` and `${{ secrets.GH_CONTAINER_REGISTRY_TOKEN }}`

change ${{ env.DOCKERHUB_REPOSITORY }} to ${{ env.DOCKER_REPOSITORY }}

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3276.london.cloudapps.digital>

File: .github/workflows/deploy_app.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
         required: true
 
 env:
- DOCKERHUB_REPOSITORY: dfedigital/teaching-vacancies
+ DOCKERHUB_REPOSITORY: ghcr.io/dfe-digital/teaching-vacancies
```
Comment (csutter) [inline .github/workflows/deploy_app.yml:14]: Should we rename this to `DOCKER_REPOSITORY`, or does something other than this workflow depend on it being called `DOCKERHUB_REPOSITORY`?

File: terraform/app/terraform.tf [hunk 2]
```diff
@@ -71,8 +71,8 @@ module "paas" {
   app_env_values                    = local.paas_app_env_values
   app_start_timeout                 = var.paas_app_start_timeout
   app_stopped                       = var.paas_app_stopped
-  docker_username                   = local.infra_secrets.docker_username
-  docker_password                   = local.infra_secrets.docker_password
+  docker_username                   = local.infra_secrets.docker_username_git
```
Comment (csutter) [inline terraform/app/terraform.tf:74]: I think `_github` would be more descriptive than `_git`

**Review by csutter — APPROVED** [review]
See comments, and there's some stray blank lines, other than that looks good 👍

**Review by saliceti — APPROVED** [review]

**Pull request merged**